### PR TITLE
Add --abandon and --yes flags to svcat deprovision command

### LINKED ISSUE #2268: Add command to svcat to abandon instances
Sometimes a service instance can become "stuck", and we want to just give up on it and make it go away in Kubernetes so that we don't see it anymore.

There are a couple ways this could happen, and I'm happy to go into more detail if whoever picks up this issue needs it. The short answer is that it can happen due to bugs, misbehaved brokers, messed up developer environments, or cases where you want to disconnect k8s from a service provisioned by the broker, even though it's still being used elsewhere (so the broker isn't letting you delete it).

This feature is to add the `--abandon` boolean flag to the [svcat unbind](https://github.com/kubernetes-incubator/service-catalog/blob/master/cmd/svcat/binding/unbind_cmd.go) and [svcat deprovision](https://github.com/kubernetes-incubator/service-catalog/blob/master/cmd/svcat/instance/deprovision_cmd.go) commands. When that flag is present, before deleting the instance, the cli will first confirm the action and then remove ALL finalizers from the service instance so that it is deleted immediately, without giving service catalog a chance to try to delete it with the broker.

It also adds an optional `-y` flag that skips the interactive prompt/confirmation.

```
$ svcat unbind --abandon INSTANCE_NAME
This action is not reversible and may cause you to be charged for the broker resources that are abandoned.
Are you sure? [y|N]

deleted BINDING_NAME(S)
deleted INSTANCE_NAME

$ svcat deprovision --abandon INSTANCE_NAME
This action is not reversible and may cause you to be charged for the broker resources that are abandoned.
Are you sure? [y|N]

deleted INSTANCE_NAME

$ svcat help deprovision
Deletes an instance of a service

Usage:
  svcat deprovision NAME [flags]

Examples:

  svcat deprovision wordpress-mysql-instance


Flags:
      --abandon bool    Forcefully and immediately delete the resource from Service Catalog ONLY, potentially abandoning any broker resources that you may continue to be charged for.
      --yes,-y bool         Automatic yes to prompts. Assume "yes" as answer to all prompts and run non-interactively.
```

* See the original issue for context and the discussion that lead up to this issue, #1723 
* I wrote an [example plugin that does this for pods](https://github.com/carolynvs/kubectl-kill-plugin/blob/7fa39d6bf5620e7bcb590611f6610e594ff5920d/main.go#L57-L67) that you can use as a reference.
* Add documentation for the feature to the [cli page on our website](https://github.com/kubernetes-incubator/service-catalog/blob/master/docs/cli.md)

### PR DESCRIPTION
**Pull request by taragu**
This PR is a 
 - [x] Feature Implementation
 - [ ] Bug Fix
 - [ ] Documentation

**What this PR does / why we need it**: Add --abandon and --yes flags to svcat deprovision command

**Which issue(s) this PR fixes** 

Fixes #2268

Please leave this checklist in the PR comment so that maintainers can ensure a good PR.

Merge Checklist:
 - [ ] New feature 
   - [ ] Tests
   - [ ] Documentation
 - [ ] SVCat CLI flag
 - [ ] Server Flag for config
   - [ ] Chart changes
   - [ ] removing a flag by marking deprecated and hiding to avoid
         breaking the chart release and existing clients who provide a
         flag that will get an error when they try to update

### CODE REVIEW

File: cmd/svcat/instance/deprovision_cmd.go [hunk 1]
```diff
@@ -68,7 +88,30 @@ func (c *deprovisonCmd) Run() error {
...
+	var err error
+	if c.abandon {
+		fmt.Fprintln(c.Output, "This action is not reversible and may cause you to be charged for the broker resources that are abandoned.")
+		if !c.skipPrompt {
+			fmt.Fprintln(c.Output, "Are you sure? [y|n]: ")
+			s := bufio.NewScanner(os.Stdin)
+			s.Scan()
+
+			err = s.Err()
+			fmt.Fprintln(c.Output, err)
```
Comment (jberkhahn) [inline cmd/svcat/instance/deprovision_cmd.go]: Are you sure you need to always print this?

**Comment by jberkhahn** [comment]
hmm, this doesn't seem like it works. When I try to get it to deprovision an instance to a broker that I've removed on the back end, the instance says the deprovision call failed and gets stuck like this:
```
apiVersion: servicecatalog.k8s.io/v1beta1
kind: ServiceInstance
metadata:
  creationTimestamp: 2019-04-05T23:15:32Z
  deletionGracePeriodSeconds: 0
  deletionTimestamp: 2019-04-05T23:18:29Z
  finalizers:
  - kubernetes-incubator/service-catalog
  generation: 2
  name: foobar
  namespace: default
  resourceVersion: "3321"
  selfLink: /apis/servicecatalog.k8s.io/v1beta1/namespaces/default/serviceinstances/foobar
  uid: b60f4476-57f8-11e9-a39a-0242ac110005
spec:
  clusterServiceClassExternalName: mysql
  clusterServiceClassRef:
    name: mysql
  clusterServicePlanExternalName: 5-7-14
  clusterServicePlanRef:
    name: mysql-5-7-14
  externalID: b60f43e7-57f8-11e9-a39a-0242ac110005
  parameters: {}
  updateRequests: 0
  userInfo:
    groups:
    - system:masters
    - system:authenticated
    uid: ""
    username: minikube-user
status:
  asyncOpInProgress: false
  conditions:
  - lastTransitionTime: 2019-04-05T23:18:52Z
    message: 'Error deprovisioning, ClusterServiceClass (K8S: "mysql" ExternalName:
      "mysql") at ClusterServiceBroker "minibroker": Delete http://minibroker-minibroker.minibroker.svc.cluster.local/v2/service_instances/b60f43e7-57f8-11e9-a39a-0242ac110005?accepts_incomplete=true&plan_id=mysql-5-7-14&service_id=mysql:
      net/http: request canceled while waiting for connection (Client.Timeout exceeded
      while awaiting headers)'
    reason: DeprovisionCallFailed
    status: Unknown
    type: Ready
  currentOperation: Deprovision
  deprovisionStatus: Required
  externalProperties:
    clusterServicePlanExternalID: mysql-5-7-14
    clusterServicePlanExternalName: 5-7-14
    userInfo:
      groups:
      - system:masters
      - system:authenticated
      uid: ""
      username: minikube-user
  inProgressProperties:
    clusterServicePlanExternalID: mysql-5-7-14
    clusterServicePlanExternalName: 5-7-14
    userInfo:
      groups:
      - system:masters
      - system:authenticated
      uid: ""
      username: minikube-user
  observedGeneration: 2
  operationStartTime: 2019-04-05T23:16:50Z
  orphanMitigationInProgress: false
  provisionStatus: Provisioned
  reconciledGeneration: 1
```

Is there some other thing we have to do to get it to delete the instance?

**Comment by jberkhahn** [comment]
Hmm, seems removing the finalizers gets us past this check: https://github.com/kubernetes-incubator/service-catalog/blob/master/pkg/controller/controller_instance.go#L825 but it will still try to reach out to the broker to deprovision. I'm not really sure what we want to do in this case. Will bring it up in the meeting on Monday.

**Comment by jboyd01** [comment]
Adding a work in progress label, please remove it once you are set on this and ready for review.  Thanks @taragu

**Comment by jberkhahn** [comment]
so just for comparison, here's the reconcile for deleting a binding: https://github.com/kubernetes-incubator/service-catalog/blob/master/pkg/controller/controller_binding.go#L348

I'm not seeing offhand how this succeeds while the deprovision case doesn't. Hmm. Maybe I'm doing something wrong on my end? I tested this by deploying a minibroker, registering it with service catalog, provisioning an instance, and then deleting the minibroker deployment directly, and then trying to svcat deprovision --abandon the instance.

File: cmd/svcat/instance/deprovision_cmd.go [hunk 2]
```diff
@@ -68,7 +88,30 @@ func (c *deprovisonCmd) Run() error {
...
+
+			if strings.ToLower(s.Text()) != "y" {
+				err = fmt.Errorf("aborted abandon operation")
+				return err
+			}
+		}
+
+		si := &v1beta1.ServiceInstance{ObjectMeta: metav1.ObjectMeta{Name: c.instanceName, Namespace: c.Namespace}}
+		if _, err = c.App.RemoveBindingFinalizerByInstance(si); err != nil {
```
Comment (jboyd01) [inline cmd/svcat/instance/deprovision_cmd.go]: As we discussed in SIG meeting on Monday, this is the problem.  We need to remove the finalizers from this **instance**.  This call to RemoveBindingFinalizerByInstance is removing the finalizer from any **bindings** related to the instance.

**Pull request merged**